src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/MonstersIncListener.java

```diff
@@ -0,0 +1,83 @@
+package com.gmail.val59000mc.scenarios.scenariolisteners;
+
+import com.gmail.val59000mc.UhcCore;
+import com.gmail.val59000mc.scenarios.ScenarioListener;
+
+import java.util.List;
+import java.util.HashMap;
+
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Bisected;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.metadata.FixedMetadataValue;
+import org.bukkit.metadata.MetadataValue;
+
+public class MonstersIncListener extends ScenarioListener {
+
+    UhcCore plugin = UhcCore.pl;
+    private int DoorsPlaced;
+    private HashMap<Integer, Location> DoorLocs = new HashMap<>();
+
+    @EventHandler (ignoreCancelled = true)
+    public void onBlockPlace(BlockPlaceEvent e) {
+
+        Block block = e.getBlock();
+        Location loc = e.getBlock().getLocation();
+
+        if(!block.getType().toString().contains("TRAP") && block.getType().toString().contains("DOOR")) {
+            block.setMetadata("DoorNum", new FixedMetadataValue(plugin, DoorsPlaced));
```

Comment:
Why do doors need a number?

Reply:
The intended behaviour is for a door to take you to a different random door. I used metadata - a door number to verify whether the door that was chosen is the same one or not. Unfortunately, I seem to be missing something with the metadata as sometimes it does not register and the player can end up teleported to the door they're using. If you know what might be causing this, I'd love insight.

If it can't be fixed, I can remove door numbers completely.

Reply:
Yea, not sure how the meta data works but I'm pretty sure it's also version specific. So I'd just remove the door numbers and use `location.equals(otherLocation);` to verify it's another door.

Reply:
Using location over metadata as of 691e7d4